src/openkit-shared/Core/Configuration/OpenKitType.cs

```diff
@@ -19,16 +19,18 @@ namespace Dynatrace.OpenKit.Core.Configuration
     public class OpenKitType
     {
 
-        public static readonly OpenKitType APPMON = new OpenKitType("dynaTraceMonitor", 1);         // AppMon: default monitor URL name contains "dynaTraceMonitor" and default Server ID is 1
-        public static readonly OpenKitType DYNATRACE = new OpenKitType("mbeacon", -1);              // Dynatrace: default monitor URL name contains "mbeacon" and default Server ID is -1
+        public static readonly OpenKitType APPMON = new OpenKitType("dynaTraceMonitor", 1, "APPMON");         // AppMon: default monitor URL name contains "dynaTraceMonitor" and default Server ID is 1
+        public static readonly OpenKitType DYNATRACE = new OpenKitType("mbeacon", -1, "DYNATRACE");              // Dynatrace: default monitor URL name contains "mbeacon" and default Server ID is -1
 
         private string defaultMonitorName;
         private int defaultServerID;
+        private string typeName;

```

Comment:
Make everything up here readonly.
It's initialized in the ctor and as far as i can see, not modified